crates/autopilot/src/limit_order_quoter.rs

```diff
@@ -0,0 +1,94 @@
+use crate::database::Postgres;
+use anyhow::Result;
+use chrono::Duration;
+use model::{
+    order::OrderKind,
+    quote::{default_verification_gas_limit, OrderQuoteSide, QuoteSigningScheme, SellAmount},
+    signature::Signature,
+};
+use shared::order_quoting::{OrderQuoting, QuoteParameters};
+use std::sync::Arc;
+
+/// Background task which quotes all limit orders and sets the surplus_fee for each one
+/// to the fee returned by the quoting process. If quoting fails, the corresponding
+/// order is skipped.
+pub struct LimitOrderQuoter {
+    pub limit_order_age: Duration,
+    pub loop_delay: std::time::Duration,
+    pub quoter: Arc<dyn OrderQuoting>,
+    pub database: Postgres,
+}
+
+impl LimitOrderQuoter {
+    pub fn spawn(self) {
+        tokio::spawn(async move {
+            loop {
+                if let Err(err) = self.update().await {
+                    tracing::error!(?err, "failed to update limit order surplus");
+                }
+                tokio::time::sleep(self.loop_delay).await;
+            }
+        });
+    }
+
+    async fn update(&self) -> Result<()> {
+        let orders = self
+            .database
+            .limit_orders_with_outdated_fees(self.limit_order_age)
+            .await?;
```

Comment:
Do you think it makes sense to have a metric of the number of orders we consider outdated?
Then we could see if back pressure is building up when the update logic is not fast enough to keep up.
Shouldn't be tackled in this PR to avoid scope creep.